kratos/testing/tester.cpp

```diff
@@ -228,8 +228,7 @@ namespace Kratos
 		{
 			// creating the regex pattern replacing * with ".*"
 
-#if defined(__GNUC__) &&( __GNUC__ < 4 || \
-              (__GNUC__ == 4 && (__GNUC_MINOR__ < 9))) 
+#if __cplusplus <= 199711L
```

Comment:
I think that this particular feature (regex) was still broken on relatively recent gccs, anyway Pooyan will remember more about this